CRM/Mailing/Form/Group.php

```diff
@@ -239,12 +239,13 @@ public function buildQuickForm() {
     //when the context is search add base group's.
     if ($this->_searchBasedMailing) {
       //get the static groups
-      $staticGroups = CRM_Core_PseudoConstant::staticGroup(FALSE, 'Mailing');
+      //CRM-16600 Include Smart Groups in Unsubscribe list as that matches
+      //all other practices in CiviMail
       $this->add('select', 'baseGroup',
         ts('Unsubscription Group'),
         array(
           '' => ts('- select -'),
-        ) + CRM_Contact_BAO_Group::getGroupsHierarchy($staticGroups, NULL, '&nbsp;&nbsp;', TRUE),
+        ) + $groups,
```

Comment:
I think it would still be nice to present the hierarchy.
So this line could be 

    ) + CRM_Contact_BAO_Group::getGroupsHierarchy($groups, NULL, '&nbsp;&nbsp;', TRUE),